aquadoggo/src/network/service.rs

```diff
@@ -140,13 +110,34 @@ pub async fn network_service(
                 } => {
                     debug!("ConnectionClosed: {peer_id} {endpoint:?} {num_established} {cause:?}")
                 }
+                SwarmEvent::ConnectionEstablished { peer_id, .. }
+                    if network_config.rendezvous_client
+                        && network_config.rendezvous_peer_id.unwrap() == peer_id =>
+                {
+                    debug!(
+                        "Connected to rendezvous point, discovering nodes in '{NODE_NAMESPACE}' namespace ..."
+                    );
+
+                    swarm
+                        .behaviour_mut()
+                        .rendezvous_client
+                        .as_mut()
+                        .unwrap()
```

Comment:
What could happen here?

Reply:
Actually, I found a way to make this slightly cleaner by using `if let` to avoid the `is_enabled()` check and `unwrap()` calls.